### DISCUSSION TITLE
how can I merge surface's together ?

### DISCUSSION THREAD

**Discussion by damavand1** [introlab/rtabmap#1076] [General]
hi

how can I merge surface's (RGBD frames) together ?
it can reduce points and increase speed

for example in this picture merge number one and number two together.

![Screenshot 2023-06-30 220118](https://github.com/introlab/rtabmap/assets/104978921/9d1db20e-32d2-49f2-933f-c95b8c46c4fe)

you know, some time scanning and environment may scan some surfaces multiple times for get more details, and it causes confusion of surfaces like image i sent you.
In other word, how can i optimize my 3D map ?

How can i solve it ?

![20230701_173054](https://github.com/introlab/rtabmap/assets/104978921/a2420092-3192-4e92-8dde-e6f40a1e1341)

thanks for you grate tool "RTAB-map"

**Comment by matlabbe** [comment]
What you are looking is a Truncated Signed Distance Function (TSDF), which rtabmap doesn't do. rtabmap can combine point clouds and generate surface (with poisson surface estimation) in post-processing, not in real-time.

One problem when doing TSDF online is that open source approaches that you may find may not handle online loop closure optimization. From the approach you are proposing, on loop closure, it may generate gaps or overlap again some points.

I did experiment with TSDF before, we can see remaining options are still seen in export dialog:
![2023-07-01_16-57](https://github.com/introlab/rtabmap/assets/2319645/975998c6-5500-4107-882c-690b09ff2ef1)

but never thought that would ever be faster to adjust the points of the map after a loop closure than just adjusting the poses of the individual points clouds in a 3D rendering engine.
